### ISSUE TITLE
Local tags should be synced when same org

### ISSUE THREAD

**Issue by eCrimeLabs** [MISP/MISP#5362]
I would be really nice to be able to sync local set tags between MISP instances that belongs to same Organisation.

Worst case In the situation where a MISP instance gets migrated by Syncing data the local tages will disappear.

**Comment by RichieB2B** [comment]
Absolutely. When a local tag is marked exportable it should be synced to remote servers with the same MISP.host_org_id

In essence there are 4 types of tags:

1. global and exportable: always synced
1. global and not exportable: never synced (enforced in tag definition)
1. local and exportable: synced to local servers
1. local and not exportable: never synced (enforced per tag use)

The difference between `exportable` and `local` is that `exportable` is a property of the tag definition while `local` is a property of the specific tag on an event or attribute. Also, `local` tags do not modify the timestamp and event publish status.

Currently type 3 does not work which I consider a bug in the local tag implementation.

**Comment by iglocska** [comment]
correct, that sounds like a bug (omission in the design).

**Comment by eCrimeLabs** [comment]
I don't know if this is covered in the @RichieB2B comment, however from the usage I've seen the local tags be seen as both an local instance tag and a local organisations tag, as mentioned it does not modify existing events but help the organisation to add context and work flow on events or attributes from data delivered by other organisations.

I should be possible to add local tags that are synced between MISP instances where the organisation is the same, meaning on the Sync side -> "Internal instance" = checked, "Organisation type" = "Local organisation" AND "local Organisation" = the same.

In the cases of using the data for internal data flow control, I would not want a tag to be exportable, it should be as "private" as possible to avoid it by mistake to be exported to external instances.

**Comment by RichieB2B** [comment]
You are describing tag type 3 and 4 from my previous comment.

**Comment by thamamo** [comment]
Hello @RichieB2B, @iglocska, @eCrimeLabs,  

I have some questions.
Would you tell me? 
(Excuse me for the stupid question.)

**Question1.** Do local tags only make sense for synchronize **between internal instances?**
In other words, 
there is no point in using the local tag in situations where there is only synchronization with the external server. 
Does that mean that the global tag should be used in such cases?

**Question2.** When I pulled an event tagged with type 3 from the external server, 
**it changed to the global tag** on the external server.
> Type 3. local and exportable

Is this behavior correct as expected? 

**Question3.** When should I use synchronization between internal servers?
In what cases do you sync your internal servers?

**Question4.** In the version I'm using (2.4.134), 
I don't see the "internal instance" checkbox when adding a server. 
Even if the Organization type is set to Local organization.
Do I have to change the settings somewhere else? 
Please tell me how to set it.

Best regards,
thamamo.

**Comment by thamamo** [comment]
Hello @RichieB2B, @iglocska, @eCrimeLabs, 

How about the above question?

I’m sorry to bother you with this question.

Best regards,
thamamo.

**Comment by Wachizungu** [comment]
I believe this can be closed

Adding references (MISP v 2.4.130 but a bunch of fixes later as well):
https://github.com/MISP/MISP/commit/17c46a8b254e129a207e272d3d5f6b02872898be

https://github.com/MISP/MISP/pull/4836
